app/controllers/tools_controller.rb

```diff
@@ -38,7 +38,24 @@ def set_body_id
   end
 
   def set_tools
-    @tools = tool_class.for_index(published_at: :desc).page(params[:page]).per(10)
+    @tools = tool_class.for_index(**filters).page(params[:page]).per(10)
+  end
+
+  def filters
+    {}.merge(sort)
+      .merge(language_filter).compact
```

Comment:
```suggestion
      .merge(language_filter)
      .compact
```

What do you think of this?